Fixes/ Review readme file (#1)
wilsoniovlabs/relaying-services-sdk-dapp: README.md

# RIF Relaying Services SDK sample dApp

> This a basic sample dApp to showcase how users can submit transactions to the RSK blockchain paying a RelayServer for the transaction fees with tokens using RIF Relay SDK. You will need to connect to the dApp with Metamask but only for signing transactions with the account that controls your Smart Wallets.

Reviewer: ```suggestion
This a basic sample dApp to showcase how users can submit transactions to the RSK blockchain paying a Relay Server for the transaction fees with tokens using RIF Relay SDK. You will need to connect to the dApp with Metamask but only for signing transactions with the account that controls your Smart Wallets.
```

Author: Fixed

> See https://github.com/rsksmart/rif-relay for the RIF Relay project.

Reviewer: it'd be good to format links in the standard markdown style (see [here](https://www.markdownguide.org/basic-syntax/#links))

Author: Fidex

> ## Using the dApp

Reviewer: ```suggestion
```

this section isn't about running the app so i would do away with this title

> ## Pre-Requisites

Reviewer: it'd be nice to specify which node version we're using

Reviewer: @wilsoniovlabs please correct me if i'm wrong, but this still has not been provided

> ## Pre-Requisites

Reviewer: is a relay server not needed to run this dapp? (asking because i am not sure)

Reviewer: correct me if i'm wrong, but aren't we still missing an indication that the server must be running?

Author: Good Catch,

Reviewer: @wilsoniovlabs sorry, maybe i am missing something, but where is this clearly indicated?

> * RSKj Node Running

Reviewer: a link here would be useful

Reviewer: @wilsoniovlabs you still have not provided the link as far as i've seen

* [Enveloping Contracts](https://github.com/anarancio/rif-relay-contracts) deployed.

> * Current Token Allowed

Reviewer: we'll need to at least provide a link for instructions on how to do this

Author: Fixed.

# RIF Relaying Services SDK sample dApp

This a basic sample dApp to showcase how users can submit transactions to the RSK blockchain paying a Relay Server for the transaction fees with tokens using RIF Relay SDK. You will need to connect to the dApp with Metamask but only for signing transactions with the account that controls your Smart Wallets.

See [RIF Relay](https://github.com/rsksmart/rif-relay) project.

## Pre-Requisites

> * [NodeJS Version v12.18.3 or higher](https://yarnpkg.com/getting-started/install).

Reviewer: this doens't look right. @antomor mentioned that he was using node v7, did you check in the terminal which version you were running?

also, this seems to be a link to yarn. i don't think a link here was necessary, although i would have added the recommendation to use nvm instead

Reviewer: still pending

* [RSKj Node Running](https://github.com/rsksmart/rskj).
* [RIF Relay Contract Contracts](https://github.com/anarancio/rif-relay-contracts) deployed.
* [Current Token Allowed](https://github.com/anarancio/rif-relay-contracts#allowing-tokens).

> * [RIF Relay Server](https://github.com/infuy/rif-relay-server) running and register.

Reviewer: ```suggestion
* [RIF Relay Server](https://github.com/infuy/rif-relay-server) running and registered.
```

Reviewer: still pending

To run the sample dApp follow these steps:

> 1. Edit the sample dApp config file located in `.env` and add the contract addresses from the deployment executed in step 2 from Running a Relay Server. By default the project imports `.env` on the `npm start`. Look for these lines and modify them accordingly:

Reviewer: @wilsoniovlabs can you please explain what step 2 refers to here?

To run the sample dApp follow these steps:

1. Edit the sample dApp config file located in `.env` and add the contract addresses. By default the project imports `.env` on the `npm start`. Look for these lines and modify them accordingly:
```
REACT_APP_CONTRACTS_RELAY_HUB=0xE0825f57Dd05Ef62FF731c27222A86E104CC4Cad
REACT_APP_CONTRACTS_DEPLOY_VERIFIER=0x1eD614cd3443EFd9c70F04b6d777aed947A4b0c4
REACT_APP_CONTRACTS_RELAY_VERIFIER=0x5159345aaB821172e795d56274D0f5FDFdC6aBD9
REACT_APP_CONTRACTS_SMART_WALLET_FACTORY=0x03F23ae1917722d5A27a2Ea0Bcc98725a2a2a49a
REACT_APP_CONTRACTS_SMART_WALLET=0x73ec81da0C72DD112e06c09A6ec03B5544d26F05
REACT_APP_CONTRACTS_RIF_TOKEN=0x726ECC75d5D51356AA4d0a5B648790cC345985ED
REACT_APP_CONTRACTS_RELAY_WORKER=0x3d67f029f778a088904f12d030933967d220faa3

REACT_APP_RIF_RELAY_CHAIN_ID=33
REACT_APP_RIF_RELAY_GAS_PRICE_FACTOR_PERCENT=0

> REACT_APP_RIF_RELAY_RELAY_LOOKUP_WINDOW_BLOCKS=1e5

Reviewer: here too

To run the sample dApp follow these steps:

> 1. Edit the sample dApp config file located in `.env` and add the contract addresses from the deployment executed in step 2 from Running a RelayServer. By default the project imports `.env` on the `npm start`. Look for these lines and modify them accordingly:

Reviewer: what does "step 2" refer to here?

> 1. Edit the sample dApp config file located in `.env` and add the contract addresses from the deployment executed in step 2 from Running a RelayServer. By default the project imports `.env` on the `npm start`. Look for these lines and modify them accordingly:

Reviewer: ```suggestion
1. Edit the sample dApp config file located in `.env` and add the contract addresses from the deployment executed in step 2 from Running a Relay Server. By default the project imports `.env` on the `npm start`. Look for these lines and modify them accordingly:
```

```
REACT_APP_CONTRACTS_RELAY_HUB=0xE0825f57Dd05Ef62FF731c27222A86E104CC4Cad
REACT_APP_CONTRACTS_DEPLOY_VERIFIER=0x1eD614cd3443EFd9c70F04b6d777aed947A4b0c4
REACT_APP_CONTRACTS_RELAY_VERIFIER=0x5159345aaB821172e795d56274D0f5FDFdC6aBD9
REACT_APP_CONTRACTS_SMART_WALLET_FACTORY=0x03F23ae1917722d5A27a2Ea0Bcc98725a2a2a49a
REACT_APP_CONTRACTS_SMART_WALLET=0x73ec81da0C72DD112e06c09A6ec03B5544d26F05

> REACT_APP_CONTRACTS_RIF_TOKEN=0x726ECC75d5D51356AA4d0a5B648790cC345985ED

Reviewer: although these are known for us, it'd be good to specify what each of them mean after this code snippet

Reviewer: @wilsoniovlabs could you please add an explanation for the meaning of each of these variables? thanks

Reviewer: see [here](https://github.com/wilsoniovlabs/relaying-services-sdk-dapp/pull/1#discussion_r742050519)

> REACT_APP_CONTRACTS_TEST_RECIPIENT=0xCd5805d60Bbf9Afe69a394c2BDa10F6Dae2c39AF

Reviewer: could you please explain what this one is used for? what is the impact of using this address instead of another?

Reviewer: @wilsoniovlabs same here, an explanation would be helpful

Reviewer: see [here](https://github.com/wilsoniovlabs/relaying-services-sdk-dapp/pull/1#discussion_r742050519)

> REACT_APP_CONTRACTS_RELAY_WORKER=0x3d67f029f778a088904f12d030933967d220faa3

Reviewer: is it possible to actually set up the worker address with this app? in all other usages i remember reading this address (through the `getAddr` API) but not setting it

Author: We can get this address from the server using getAddr, but i think that we can add this change for the SDK and remove this parameter from the .env configuration.

Reviewer: sorry, but i'm not sure what the flow is supposed to be here.

are we supposed to get this value from the `getAddr` endpoint (or similar) and then put it here?

or is the value written here going to be set as the worker address?

Author: get this value from the getAddr endpoint (or similar) and then put it here.

Reviewer: what about the test recipient?

see [here](https://github.com/wilsoniovlabs/relaying-services-sdk-dapp/pull/1#discussion_r728562665)

Author: It is not necessary to use the SDK, I think  that we can get all of this address from the server using the SDK to get more easy integration.

Reviewer: @wilsoniovlabs ok, so should we remove this? sorry, it's not clear to me yet

REACT_APP_ENVELOPING_CHAIN_ID=33
REACT_APP_ENVELOPING_GAS_PRICE_FACTOR_PERCENT=0

> REACT_APP_ENVELOPING_RELAY_LOOKUP_WINDOW_BLOCKS=1e5

Reviewer: could you please explain what these variables mean?

Author: This variables is override on the SDK, it will remove.

Reviewer: @wilsoniovlabs sorry, i don't understand what you mean. could you please explain this in another way? thanks

Reviewer: @wilsoniovlabs has agreed to create and work on an issue to document these variables in the SDK documentation/integration manual

> 4. Open a browser and navigate to http://localhost:3000

Reviewer: are you sure this port is the one we use? i haven't been able to find it in the code

Author: Yes, i'm sure. this is the react-scripts default port.

Reviewer: gotcha, thanks :+1:
